docs/communicate-with-javascript.md

```diff
@@ -550,6 +550,71 @@ let () = match [%bs.external __filename] with
 | None -> Js.log "non-node environment"
 ```
 
+## Inlining constant values
+
+Some JavaScript idioms require special constants to be inlined since they serve
+as de-facto directives for bundlers. A common example is `process.env.NODE_ENV`:
+
+```js
+if(process.env.NODE_ENV !== 'production') {
+   // Development-only code
+}
+```
+
+becomes:
+
+```js
+if('development' !== 'production') {
+   // Development-only code
+}
+```
+
+In this case, bundlers such as Webpack can tell that the `if` statement always
+evaluates to a specific branch and eliminate the others.
+
+Melange provides the `[@bs.inline]` construct to achieve the same goal in
+generated JavaScript. Let's look at an example:
+
+```ocaml
+external node_env : string = "NODE_ENV" [@@bs.val] [@@bs.scope "process", "env"]
+
+(* This variable will be used in the comparison below. *)
+let development = "development"
+let () = if node_env <> development then Js.log "Only in Production"
+
+(* The value of this variable will be inlined in the final output. The variable
+   won't be present in the JavaScript output. *)
+let development_inline = "development" [@@bs.inline]
+let () = if node_env <> development_inline then Js.log "Only in Production"
+```
+
+As we can see in the generated JavaScript presented below:
+
+- the `development` variable is emitted
+    - it gets used as a variable `process.env.NODE_ENV !== development` in the
+      `if` statement
+- the `development_inline` variable isn't present in the final output
+    - its value is inlined in the `if` statement: `process.env.NODE_ENV !== "development"`
+
+```js
+// Generated by Melange
+'use strict';
+
+
+var development = "development";
+
+if (process.env.NODE_ENV !== development) {
+  console.log("Only in Production");
+}
+
+if (process.env.NODE_ENV !== "development") {
+  console.log("Only in Production");
+}
+
+exports.development = development;
+/*  Not a pure module */
```

Comment:
We can skip the first lines and exports from the generated JS, to avoid some noise (I don't think they add much when presented in the docs):

```suggestion
var development = "development";

if (process.env.NODE_ENV !== development) {
  console.log("Only in Production");
}

if (process.env.NODE_ENV !== "development") {
  console.log("Only in Production");
}
```